tests/cupy_tests/indexing_tests/test_insert.py

```diff
@@ -148,3 +148,37 @@ def test_1darray(self, xp, dtype):
         a = testing.shaped_arange(5, xp, dtype)
         val = self._compute_val(xp)
         xp.fill_diagonal(a, val=val, wrap=self.wrap)
+
+
+@testing.parameterize(*testing.product({
+    'n': [2, 4],
+    'ndim': [2, 3],
+}))
+@testing.gpu
+class TestDiagIndices(unittest.TestCase):
+
+    @testing.numpy_cupy_array_equal()
+    def test_diag_indices(self, xp):
+        return xp.diag_indices(self.n, self.ndim)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 3), (2, 2, 2)],
+}))
```

Comment:
Test non-squared matrices and invalid values

Reply:
The last two tests in this class, `test_non_equal_dims` and `test_1darray` are to check for the invalid values.

Reply:
Or do you mean something else by `invalid values`?

Reply:
Are those tests okay?

Reply:
Shapes like (0,0), (3,4), (5), (-1) and so